File: example-project-dotnet48/WebApiExample.WebApp/Areas/HelpPage/Controllers/HelpController.cs
using System;
using Microsoft.AspNetCore.Mvc;
using WebApiExample.WebApp.Areas.HelpPage.ModelDescriptions;
using WebApiExample.WebApp.Areas.HelpPage.Models;

namespace WebApiExample.WebApp.Areas.HelpPage.Controllers
{
    /// <summary>
    /// The controller that will handle requests for the help page.
    /// </summary>
    [Route("api/[controller]")]
    public class HelpController : Controller
    {
        private const string ErrorViewName = "Error";

        public HelpController()
        {
        }

        [HttpGet]
        public IActionResult Index()
        {
            // Migration note: ASP.NET Core does not have GlobalConfiguration;
            // Replace with your own documentation provider logic as needed.
            ViewBag.DocumentationProvider = null;
            return View();
        }

        [HttpGet("{apiId}")]
        public IActionResult Api(string apiId)
        {
            if (!String.IsNullOrEmpty(apiId))
            {
                HelpPageApiModel apiModel = null; // TODO: Retrieve your API model
                if (apiModel != null)
                {
                    return View(apiModel);
                }
            }

            return View(ErrorViewName);
        }

        [HttpGet("model/{modelName}")]
        public IActionResult ResourceModel(string modelName)
        {
            if (!String.IsNullOrEmpty(modelName))
            {
                ModelDescription modelDescription = null; // TODO: Retrieve your model description
                if (modelDescription != null)
                {
                    return View(modelDescription);
                }
            }

            return View(ErrorViewName);
        }
    }

File: java/com/example/webapp/controller/HelpController.java
package com.example.webapp.controller;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.servlet.ModelAndView;

@RestController
@RequestMapping("/help")
public class HelpController {

    private static final String ERROR_VIEW_NAME = "Error";

    @GetMapping("/index")
    public ModelAndView index() {
        ModelAndView modelAndView = new ModelAndView();
        modelAndView.setViewName("index");
        // Add additional attributes to the model if needed
        return modelAndView;
    }

    @GetMapping("/api")
    public ModelAndView api(String apiId) {
        ModelAndView modelAndView = new ModelAndView();
        if (apiId != null && !apiId.isEmpty()) {
            // Logic to get HelpPageApiModel based on apiId
            // If found, set the view name and add the model to the view
            modelAndView.setViewName("api");
        } else {
            modelAndView.setViewName(ERROR_VIEW_NAME);
        }
        return modelAndView;
    }

    @GetMapping("/resourceModel")
    public ModelAndView resourceModel(String modelName) {
        ModelAndView modelAndView = new ModelAndView();
        if (modelName != null && !modelName.isEmpty()) {
            // Logic to get ModelDescription based on modelName
            // If found, set the view name and add the model to the view
            modelAndView.setViewName("resourceModel");
        } else {
            modelAndView.setViewName(ERROR_VIEW_NAME);
        }
        return modelAndView;
    }
}